src/main/java/com/google/devtools/build/lib/packages/RuleClass.java

```diff
@@ -706,6 +743,21 @@ public RuleClass build(String name, String key) {
       if (type == RuleClassType.PLACEHOLDER) {
         Preconditions.checkNotNull(ruleDefinitionEnvironmentHashCode, this.name);
       }
+      if (executionPlatformConstraintsAllowed == ExecutionPlatformConstraintsAllowed.NONE) {
+        Preconditions.checkArgument(
+            executionPlatformConstraints.isEmpty(),
+            "Rule %s does not allow setting execution platform constraints",
+            name);
+      } else if (executionPlatformConstraintsAllowed
+          == ExecutionPlatformConstraintsAllowed.PER_TARGET) {
+        // Only rules that allow per target execution constraints need this attribute.
+        this.add(
```

Comment:
There's no danger in this overwriting the already existing attribute in a toolchain rule?